loom/engine/bindings/loom/lmSQLite.cpp

```diff
@@ -0,0 +1,467 @@
+        {
+            lmLogError(gSQLiteGroup, "Error calling step for database: %s", parentDB->getDBName());
+        }
+        return result;
+    }
+
+    void stepAsync(lua_State *L)
+    {
+//BEN QUESTION: I need L in order to lualoom_pushnative<> the Statement into the Delegate 
+//              in the thread onComplete... but obviously it doen't always remain valid from 
+//              this call until the time for it to be used at the end of the stepAsync() thread... 
+//              what other options are there for me to pass the Statement through to the Delegate?
+//
+//              NOTE: This wont be an issue if we can nuke the Statement from needing to be passed
+//                    into the onStatementComplete delegate... which i'm still not sure why it is
+//                    done there but not for onStatementProgress...
+//
+//HACK: NOT STABLE!!!!
+stepAsyncLuaState = L;
+        loom_thread_start(Statement::stepAsyncBody, this);    
+    }
+
+
+    const char *columnName(int col)
+    {
+        return sqlite3_column_name(statementHandle, col);
+    }
+
+    int columnType(int col)
+    {
+        return sqlite3_column_type(statementHandle, col);
+    }
+
+    int columnInt(int col)
+    {
+        return sqlite3_column_int(statementHandle, col);
+    }
+
+    double columnDouble(int col)
+    {
+        return sqlite3_column_double(statementHandle, col);
+    }
+
+    const char* columnString(int col)
+    {
+        return (const char *)sqlite3_column_text(statementHandle, col);
+    }
+
+    utByteArray *columnBytes(int col)
+    {
+        //get the blob from the column
+        void *blob = (void *)sqlite3_column_blob(statementHandle, col);
+        if(blob == NULL)
+        {
+            return NULL;
+        }
+        int size = sqlite3_column_bytes(statementHandle, col);
+
+        //valid blob so allocate byte array for it
+        utByteArray *bytes = new utByteArray();
+        bytes->allocateAndCopy(blob, size);     
+
+        return bytes;   
+    }
+
+    int reset()
+    {
+        int result = sqlite3_reset(statementHandle); 
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling reset for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+    int finalize()
+    {
+        int result = sqlite3_finalize(statementHandle); 
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling finalize for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+};
+
+
+//---Statement--- external function initialisation
+int __stdcall Statement::stepAsyncBody(void *param)
+{
+    //get the statement
+    Statement *s = (Statement *)param;
+
+
+    //call the internal SQLite step function, then yield the thread
+    int result = sqlite3_step(s->statementHandle); 
+    loom_thread_yield();
```

Comment:
Make what configurable? The yield? Will that even work?
On 12 Feb 2015 10:55 PM, "Ben Garney" <notifications@github.com> wrote:

> In loom/engine/bindings/loom/lmSQLite.cpp
> <https://github.com/LoomSDK/LoomSDK/pull/261#discussion_r24619307>:
>
> > +        }
> > +        return result;
> > +    }
> > +};
> > +
> > +
> > +//---Statement--- external function initialisation
> > +int __stdcall Statement::stepAsyncBody(void *param)
> > +{
> > +    //get the statement
> > +    Statement *s = (Statement *)param;
> > +
> > +
> > +    //call the internal SQLite step function, then yield the thread
> > +    int result = sqlite3_step(s->statementHandle);
> > +    loom_thread_yield();
>
> This can have bad side effects on some devices... Do we maybe want to make
> it configurable as a param when calling the stepAsync() method from LS?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LoomSDK/LoomSDK/pull/261/files#r24619307>.
>